### DISCUSSION TITLE
[Suggestion] Improvements to /give, for example NBT data

### DISCUSSION THREAD

**Discussion by mrfloris** [Zrips/CMI#3480] [Feature Requests]
**Description of feature/improvement request:**
 I see the question come up on Discord every week or so, requesting how it is possible to make custom damaged items with /give or if CMI give can do certain things. A lot is thankfully already possible. But some combinations are not yet.. 
---
### Basic explanation as to why this feature is needed and/or useful to have: ###

Here's a random example of what i see on discord:
```
 TheRobLP: Suggestion: NBT support and NBT tab-completing for /give command . Im too tired rn to make a detailed suggestion on Github rn
TheRobLP: I meant things like adding Custom damage to swords, or custom armor rating to armor
 TheRobLP: Also, tab-complete for NBT would be a very awesome feature. It would really help reducing the time for item NBT manipulation
 TheRobLP: Of course not every NBT can be made tab-completed because some use JSON, but I think tags like Invulnerable: are easily possible to make tab-completing
```

I see people ask certain things that a custom item made with individual commands could do, including hiding enchants, or some fancy /minecraft:give tricks, but they don't seem to yield any results in cmi give, or we've yet to figure out how.

**Comment by Zrips** [comment]
You can use generic tag's for this one, like **/cmi give Zrips enchanted_book 1 {StoredEnchantments:[{id:sharpness,lvl:1}]}** other than this we would need some special format for defining NBT data on items without duplicating existing json format. 
Any suggestions?
